business/suggestionService.js

```diff
@@ -19,12 +19,30 @@ class SuggestionService {
   async get(coordinates) {
     const definitions = await this._getRelatedDefinitions(coordinates)
     if (!definitions) return null
+    let hasSuggested = false
     const suggestion = { coordinates }
-    this._collectSuggestionsForField(definitions, suggestion, 'licensed.declared', (definition, field) =>
-      isDeclaredLicense(get(definition, field))
-    )
-    this._collectSuggestionsForField(definitions, suggestion, 'described.sourceLocation')
-    return suggestion
+    if (Array.isArray(definitions.related) && definitions.related.length > 0) {
+      hasSuggested = this._collectSuggestionsForField(definitions, suggestion, 'licensed.declared', (definition, field) =>
+        isDeclaredLicense(get(definition, field))
+      )
+      hasSuggested |= this._collectSuggestionsForField(definitions, suggestion, 'described.sourceLocation')
+    }
+
+    if (!isDeclaredLicense(get(definitions.original, 'licensed.declared'))) {
+      let discoveredExpressions = get(definitions.original, 'licensed.facets.core.discovered.expressions')
+      if (Array.isArray(discoveredExpressions)) {
+        let appendDeclared = discoveredExpressions
+          .map(exp => { return { exp, cutAt: exp.indexOf('-') } })
+          .filter(marked => marked.cutAt > 0)
+          .map(marked => { return { value: marked.exp.substr(0, marked.cutAt), version: marked.exp.substr(marked.cutAt + 1) } })
+        let suggestedSoFar = get(suggestion, 'licensed.declared') || []
+        hasSuggested |= setIfValue(suggestion, 'licensed.declared', appendDeclared
+          .concat(suggestedSoFar)
+          .filter((v, i, a) => a.indexOf(v) === i))
+      }
+    }
+
+    return hasSuggested ? suggestion : null
```

Comment:
this has the affect of returning Not Found instead of the shell of a suggestion

```
{
  "coordinates": {
    "type": "nuget",
    "provider": "nuget",
    "name": "name",
    "revision": "12.0.1"
  }
}
```

I think this is a good change, do you know if the website code is setup to handle this?

Reply:
Took it from a test, because in other cases there was a null being returned already (I think it's better if we keep consistent). So I assume the website will handle this; it won't hurt to check with Simone, which I will do, though.

Reply:
👍 makes sense. thanks